### PR TITLE
add documentation, github pages publisher

### PR DESCRIPTION
**Pull request by rtertiaer**
This PR adds Google style docstrings to pyamplipi, and additionally adds a Sphinx renderer and a github action that publishes to Github Pages. I'm not gonna pretend - I had machine help with a great deal of this drudgery. However, I've reviewed and tweaked everything - builds work, and I am not aware of any spots where the API is described incorrectly. I'd love more eyes on that though. there's some follow-up work to this once the github action publishes to the gh-pages branch, but it's pretty minimal. I gave this an earnest review and it again looks pretty swell, save for some funny formatting on examples I'm not going to fuss with for now - this documentation is in support of a series of blog posts describing AmpliPi's API.

### CODE REVIEW

File: .github/workflow/sphinx-docs.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.x'
+          cache: 'pip'
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install sphinx sphinx-rtd-theme myst-parser
```
Comment (linknum23) [inline .github/workflow/sphinx-docs.yml:29]: does this belong in `requirements-doc-gen.txt`  or something similar?

File: .github/workflow/sphinx-docs.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+          if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
+          pip install .
+
+      - name: Build documentation
+        run: |
+          cd docs
+          # Generate API documentation
+          sphinx-apidoc -o . ../pyamplipi -f
```
Comment (linknum23) [inline .github/workflow/sphinx-docs.yml]: should this be part of the Makefile?

File: .github/workflow/sphinx-docs.yml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+          # Build HTML docs
+          make html
+          touch _build/html/.nojekyll
```
Comment (linknum23) [inline .github/workflow/sphinx-docs.yml:36]: is this useful? why?

**Review by linknum23 — COMMENTED** [review]
How can I make these docs and look at them on my computer?

File: .github/workflow/sphinx-docs.yml (hunk 3 above)
Comment (rtertiaer) [reply .github/workflow/sphinx-docs.yml:36]: this is machine generated; I haven't played with the why, and honestly didn't care to. This is not an axis I particularly wanted to optimize along.

Comment (rtertiaer) [reply .github/workflow/sphinx-docs.yml:36]: https://github.blog/news-insights/bypassing-jekyll-on-github-pages/

File: docs/index.rst [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+   modules
+
+
+Indices and tables
+==================
+
+* :ref:`genindex`
+* :ref:`modindex`
+* :ref:`search`
```
Comment (linknum23) [inline docs/index.rst]: From my testing the search page is useless and doesn't work. The actual sidebar search works decently. No need to link to a useless page. 
```suggestion
```

**Comment by linknum23** [comment]
One side note. On a recent version of python (3.11) `make init-dev`fails to build  (`make init` works fine)

```
pyamplipi/amplipi.py:536: error: Incompatible default for argument "timeout" (default has type "None", argument has type "int")  [assignment]
```

The fix is to update the announcement definition in `pyamplipi/amplipi.py` to 
``` async def announce(self, announcement: Announcement, timeout: Optional[int] = None) -> Status:```

File: Makefile [hunk 5]
```diff
@@ -3,7 +3,7 @@ PYTHON = python3
 TEST_PATH = ./tests/
 FLAKE8_EXCLUDE = venv,.venv,.eggs,.tox,.git,__pycache__,*.pyc
 
-.PHONY: clean init init-de test check build
+.PHONY: clean init init-de test check build docs
```
Comment (linknum23) [inline Makefile]:
```suggestion
.PHONY: clean init init-dev test check build docs
```

**Review by linknum23 — APPROVED** [review]
As a whole the docs look good. Please make the minor changes above before merging.

**Pull request merged**